connectors/sources/notion.py

```diff
@@ -176,17 +177,29 @@ async def fetch_children_recursively(block):
                     ):  # pyright: ignore
                         yield grandchild
 
-        async for block in self.async_iterate_paginated_api(
-            self._get_client.blocks.children.list, block_id=block_id
-        ):
-            if block.get("type") not in ["child_database", "child_page", "unsupported"]:
-                yield block
-                if block.get("has_children") is True:
-                    async for child in fetch_children_recursively(block):
-                        yield child
-            if block.get("type") == "child_database":
-                async for record in self.query_database(block.get("id")):
-                    yield record
+        try:
+            async for block in self.async_iterate_paginated_api(
+                self._get_client.blocks.children.list, block_id=block_id
+            ):
+                if block.get("type") not in [
+                    "child_database",
+                    "child_page",
+                    "unsupported",
+                ]:
+                    yield block
+                    if block.get("has_children") is True:
+                        async for child in fetch_children_recursively(block):
+                            yield child
+                if block.get("type") == "child_database":
+                    async for record in self.query_database(block.get("id")):
+                        yield record
+        except APIResponseError as error:
+            if error.code == "validation_error" and "external_object" in str(error):
```

Comment:
I assume APIResponseError doesn't have a property you can look at to get at the `external_object`? I don't love having to use `str()`, as that widens the opportunity for false positives.

Reply:
Updated with commit id: [fe41c9a](https://github.com/elastic/connectors/pull/2337/commits/fe41c9a2a9dbaaad1dad9cdd700ac33631c90a19)